### PR TITLE
daemon: Separate endpoint API handlers from daemon and add tests

### PR DESCRIPTION
**Pull request by joamaki**
This moves the handlers for `/endpoint` to `daemon/restapi/endpoint.go` and thus decouples them from `type Daemon struct` which reduces their dependencies and allows to easily test them.

While most of the handlers now only require access to `EndpointManager` API, the endpoint creation still need to go through `Daemon` as not everything it depends on has been modularized (it is close though!). For creation `Daemon` implements `EndpointModifier` API which the endpoint handlers can depend on.

For testing the handlers this PR also introduces the `server.TestServer` which runs the API server using `httptest`.

### CODE REVIEW

File: daemon/restapi/endpoints_test.go [hunk 1]
```diff
@@ -0,0 +1,395 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (joamaki) [inline daemon/restapi/endpoints_test.go]: TODO remember to rename this file as well

**Review by aanm — APPROVED** [review]

File: daemon/restapi/endpoint_test.go [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+	})
+	t.Run("200", func(t *testing.T) {
+		for _, epNumID := range testEndpointIDs {
+			epID := endpointid.NewCiliumID(int64(epNumID))
+			ok, err := ctx.client.EndpointGet(epID)
+			assert.NoError(t, err)
+			if assert.NotNil(t, ok) {
+				assert.EqualValues(t, epNumID, ok.ID)
+			}
+			// TODO what else to assert?
```
Comment (bimmlerd) [inline daemon/restapi/endpoint_test.go:189]: I'd vote in favour of having golden tests here; manually asserting seems more error-prone

**Review by bimmlerd — APPROVED** [review]

**Review by joestringer — CHANGES_REQUESTED** [review]
LGTM from the perspective of improving code cleanliness, code coverage in testing. There's a scattering of TODOs that look like they should be grappled with prior to merging. Once those are addressed, feel free to dismiss this review and go ahead with merging.

**Pull request closed without merging**